### PR TITLE
Feature/sugar api

### PR DESCRIPTION
**Pull request by afaikafk**
-Add `FSO.parent` property to FSO objects that returns parent path
-Add `fsal.filter` method which returns multiple FSO objects matching the given list of paths
-Fix missing count arg from descendants api when an invalid path is given

### CODE REVIEW

File: fsal/fsdbmanager.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def list_descendants(self, path, count=False, offset=None, limit=None,
                          entry_type=None, span=None, order=None, ignored_paths=None):
         d = self._get_dir(path)
         if d is None:
-            return (False, [])
+            return (False, None, [])
```
Comment (foxbunny) [inline fsal/fsdbmanager.py:139]: So... where's the sugar?

File: fsal/fs.py [hunk 2]
```diff
@@ -28,6 +28,10 @@ def base_path(self):
         return self._base_path
 
     @property
+    def parent(self):
+        return os.path.dirname(self._path)
```
Comment (foxbunny) [inline fsal/fs.py:32]: This is missing sugar.

File: fsal/fsdbmanager.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 import time
 import collections
 import functools
-from itertools import ifilter
+from itertools import chain, ifilter
```
Comment (foxbunny) [inline fsal/fsdbmanager.py:9]: sugar is not imported here

File: fsal/fsdbmanager.py [hunk 4]
```diff
@@ -166,6 +166,16 @@ def list_descendants(self, path, count=False, offset=None, limit=None,
         row_iter = self.db.fetchiter(q, filter_args)
         return (True, None, self._fso_row_iterator(row_iter))
 
+    def filter(self, paths, batch_size=999):
```
Comment (foxbunny) [inline fsal/fsdbmanager.py:169]: No sugar, and no docstrings

**Comment by afaikafk** [comment]
I figured I should be consistent with everything, including missing docs,etc :)

**Comment by foxbunny** [comment]
Well, ok, then add docstrings to _everything_, muwahahaha!

**Comment by foxbunny** [comment]
Looks good to me.

**Pull request merged**